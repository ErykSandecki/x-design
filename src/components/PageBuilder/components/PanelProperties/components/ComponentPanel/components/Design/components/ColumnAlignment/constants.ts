// others
import { translationNameSpace as parentNameSpace } from '../../constants';

// types
import { AlignmentHorizontal, AlignmentVertical } from 'types';
import { TAligmentHorizontalButtonGroup, TAligmentVerticalButtonGoup } from './types';

export const HORIZONTAL_BUTTONS: TAligmentHorizontalButtonGroup = [
  { key: AlignmentHorizontal.left, name: 'AlignHorizontalLeft' },
  { key: AlignmentHorizontal.center, name: 'AlignHorizontalCenter' },
  { key: AlignmentHorizontal.right, name: 'AlignHorizontalRight' },
];

export const VERTICAL_BUTTONS: TAligmentVerticalButtonGoup = [
  { key: AlignmentVertical.top, name: 'AlignVerticalTop' },
  { key: AlignmentVertical.center, name: 'AlignVerticalCenter' },
  { key: AlignmentVertical.bottom, name: 'AlignVerticalBottom' },
];

export const translationNameSpace = `${parentNameSpace}.columnAlignment`;
