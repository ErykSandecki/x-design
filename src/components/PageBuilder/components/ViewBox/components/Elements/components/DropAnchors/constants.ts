// types
import { DropAnchorsPosition } from './enums';

export const HORIZONTAL_ANCHORS = [
  DropAnchorsPosition.left,
  DropAnchorsPosition.right,
];
export const VERTICAL_ANCHORS = [
  DropAnchorsPosition.bottom,
  DropAnchorsPosition.top,
];
