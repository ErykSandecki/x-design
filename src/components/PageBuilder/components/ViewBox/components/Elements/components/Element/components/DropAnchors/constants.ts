// types
import { DropAnchorsPosition } from 'store/pageBuilder/enums';

export const HORIZONTAL_ANCHORS = [
  DropAnchorsPosition.left,
  DropAnchorsPosition.right,
];

export const VERTICAL_ANCHORS = [
  DropAnchorsPosition.bottom,
  DropAnchorsPosition.top,
];

export const promptsData = (
  anchorPos: DropAnchorsPosition,
  displayNextPrompt: boolean,
  displayPrevPrompt: boolean,
  isFlowVertical: boolean,
  isGrid: boolean,
): Array<{ key: DropAnchorsPosition; visible: boolean }> => [
  {
    key: DropAnchorsPosition.top,
    visible:
      anchorPos === DropAnchorsPosition.top &&
      displayPrevPrompt &&
      (isGrid || isFlowVertical),
  },
  {
    key: DropAnchorsPosition.left,
    visible:
      anchorPos === DropAnchorsPosition.left &&
      displayPrevPrompt &&
      (isGrid || !isFlowVertical),
  },
  {
    key: DropAnchorsPosition.bottom,
    visible:
      anchorPos === DropAnchorsPosition.bottom &&
      displayNextPrompt &&
      (isGrid || isFlowVertical),
  },
  {
    key: DropAnchorsPosition.right,
    visible:
      anchorPos === DropAnchorsPosition.right &&
      displayNextPrompt &&
      (isGrid || !isFlowVertical),
  },
];
