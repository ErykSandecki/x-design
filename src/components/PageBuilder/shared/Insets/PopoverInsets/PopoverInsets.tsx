import { FC } from 'react';
import { useDispatch } from 'react-redux';
import { useTranslation } from 'react-i18next';

// components
import { TIconProps, UITools } from 'shared';

// store
import { applyElementsInsetType } from 'store/pageBuilder/actions';

// types
import { PopoverItem } from '../enums';
import { TInsets, TInsetsName } from 'types';

// utils
import { isPureNumber } from 'utils';

const { PopoverCompound } = UITools;

export type TPopoverInsetsProps = {
  icon: TIconProps['name'];
  insetsName: TInsetsName;
  insets: Array<keyof TInsets>;
  isMixed: boolean;
  translationNameSpace: string;
  value: string;
};

const PopoverInsets: FC<TPopoverInsetsProps> = ({ icon, insetsName, insets, isMixed, translationNameSpace, value }) => {
  const key = insets.join('');
  const dispatch = useDispatch();
  const { t } = useTranslation();

  return (
    <>
      <PopoverCompound.PopoverItem
        icon={icon}
        index={PopoverItem.fixed}
        onClick={() => dispatch(applyElementsInsetType(insets, insetsName, 'fixed'))}
        selected={!isMixed && isPureNumber(value)}
        text={t(`${translationNameSpace}.popover.${key}.1`, { value })}
      />
    </>
  );
};

export default PopoverInsets;
