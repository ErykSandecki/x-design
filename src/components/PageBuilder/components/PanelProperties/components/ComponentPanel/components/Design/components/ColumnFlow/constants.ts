// others
import { LayoutType } from 'types';
import { translationNameSpace as parentNameSpace } from '../../constants';

// types
import { TToggleButton } from 'shared/UITools/components/ToggleButtonGroup/types';

export const translationNameSpace = `${parentNameSpace}.columnFlow`;

export const toggleButtons: Array<TToggleButton<LayoutType>> = [
  { icon: 'FlowDefault', value: LayoutType.default },
  { icon: 'FlowVertical', value: LayoutType.vertical },
  { icon: 'FlowHorizontal', value: LayoutType.horizontal },
  { icon: 'FlowGrid', value: LayoutType.grid },
];
