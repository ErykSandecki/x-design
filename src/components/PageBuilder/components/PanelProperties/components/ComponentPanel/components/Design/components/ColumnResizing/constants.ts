// others
import { translationNameSpace as parentNameSpace } from '../../constants';

// types
import { TPopoverItem } from 'shared/UITools/components/Popover/types';
import { TT } from 'types';

export const translationNameSpace = `${parentNameSpace}.columnResizing`;

export const heightPopoverData = (t: TT): Array<TPopoverItem> => [
  {
    icon: 'FixedHeight',
    selected: true,
    text: t(`${translationNameSpace}.popover.height.1`),
  },
  { icon: 'AutoHeight', text: t(`${translationNameSpace}.popover.height.2`) },
  { separator: true },
  { icon: 'MinHeight', text: t(`${translationNameSpace}.popover.height.3`) },
  { icon: 'MaxHeight', text: t(`${translationNameSpace}.popover.height.4`) },
];

export const widthPopoverData = (t: TT): Array<TPopoverItem> => [
  {
    icon: 'FixedWidth',
    selected: true,
    text: t(`${translationNameSpace}.popover.width.1`),
  },
  { icon: 'AutoWidth', text: t(`${translationNameSpace}.popover.width.2`) },
  { separator: true },
  { icon: 'MinWidth', text: t(`${translationNameSpace}.popover.width.3`) },
  { icon: 'MaxWidth', text: t(`${translationNameSpace}.popover.width.4`) },
];
