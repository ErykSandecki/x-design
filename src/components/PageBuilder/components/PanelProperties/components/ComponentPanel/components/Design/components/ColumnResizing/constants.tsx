import { Dispatch } from 'redux';
import { ReactNode } from 'react';

// components
import { UITools } from 'shared';

// others
import { translationNameSpace as parentNameSpace } from '../../constants';

// store
import { applyElementsSizeType } from 'store/pageBuilder/actions';

// types
import { TElement } from 'types';

// utils
import { isPureNumber } from 'utils';

const { PopoverCompound } = UITools;

export const translationNameSpace = `${parentNameSpace}.columnResizing`;

export const heightPopoverData = (
  dispatch: Dispatch,
  height: TElement['height'],
  isMixed: boolean,
  t: TT,
): ReactNode => (
  <>
    <PopoverCompound.PopoverItem
      icon="FixedHeight"
      index={0}
      onClick={() => dispatch(applyElementsSizeType('height', 'fixed'))}
      selected={!isMixed && isPureNumber(height.value)}
      text={t(`${translationNameSpace}.popover.height.1`)}
    />
    <PopoverCompound.PopoverItem
      icon="AutoHeight"
      index={1}
      onClick={() => dispatch(applyElementsSizeType('height', 'auto'))}
      selected={!isMixed && height.value === 'auto'}
      text={t(`${translationNameSpace}.popover.height.2`)}
    />
    <PopoverCompound.PopoverSeparator />
    <PopoverCompound.PopoverItem
      icon="MinHeight"
      index={2}
      selected={false}
      text={t(`${translationNameSpace}.popover.height.3`)}
    />
    <PopoverCompound.PopoverItem
      icon="MaxHeight"
      index={3}
      selected={false}
      text={t(`${translationNameSpace}.popover.height.4`)}
    />
  </>
);

export const widthPopoverData = (
  dispatch: Dispatch,
  isMixed: boolean,
  t: TT,
  width: TElement['height'],
): ReactNode => (
  <>
    <PopoverCompound.PopoverItem
      icon="FixedWidth"
      index={0}
      onClick={() => dispatch(applyElementsSizeType('width', 'fixed'))}
      selected={!isMixed && isPureNumber(width.value)}
      text={t(`${translationNameSpace}.popover.width.1`)}
    />
    <PopoverCompound.PopoverItem
      icon="AutoWidth"
      index={1}
      onClick={() => dispatch(applyElementsSizeType('width', 'auto'))}
      selected={!isMixed && width.value === 'auto'}
      text={t(`${translationNameSpace}.popover.width.2`)}
    />
    <PopoverCompound.PopoverSeparator />
    <PopoverCompound.PopoverItem
      icon="MinWidth"
      index={2}
      selected={false}
      text={t(`${translationNameSpace}.popover.width.3`)}
    />
    <PopoverCompound.PopoverItem
      icon="MaxWidth"
      index={3}
      selected={false}
      text={t(`${translationNameSpace}.popover.width.4`)}
    />
  </>
);
