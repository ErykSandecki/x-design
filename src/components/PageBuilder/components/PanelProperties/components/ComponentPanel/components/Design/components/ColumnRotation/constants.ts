import { Dispatch } from 'redux';
import { noop } from 'lodash';

// others
import { translationNameSpace as parentNameSpace } from '../../constants';

// store
import { rotateElements } from 'store/pageBuilder/actions';

// types
import { TButtonGroup } from 'shared/UITools/components/ButtonGroup/types';
import { TElement } from 'types';

// utils
import { normalizeAngle } from 'utils';

export const translationNameSpace = `${parentNameSpace}.columnRotation`;

export const OPTIONS_BUTTONS = (
  angle: TElement['angle'],
  dispatch: Dispatch,
): Array<TButtonGroup> => [
  {
    name: 'ToggleRotate',
    onClick: () => dispatch(rotateElements(normalizeAngle(angle + 90))),
  },
  { name: 'FlipHorizontal', onClick: noop },
  { name: 'FlipVertical', onClick: noop },
];
