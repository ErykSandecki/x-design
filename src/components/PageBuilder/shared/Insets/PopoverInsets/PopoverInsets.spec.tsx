import { noop } from 'lodash';

// components
import PopoverInsets from './PopoverInsets';
import { UITools } from 'shared';

// others
import { translationNameSpace as paddingTranslationNameSpace } from '../../../PanelProperties/ComponentPanel/Design/ColumnPadding/constants';

// utils
import { customRender } from 'test';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

describe('PopoverGap snapshots', () => {
  it('should render PopoverInsets', () => {
    // before
    const { asFragment } = customRender(
      <UITools.PopoverCompound.PopoverRoot selected={false} setSelected={noop}>
        <PopoverInsets
          icon="PaddingB"
          insets={['b', 'l', 'r', 't']}
          insetsName="padding"
          isMixed={false}
          translationNameSpace={paddingTranslationNameSpace}
          value="1"
        />
      </UITools.PopoverCompound.PopoverRoot>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with mixed value', () => {
    // before
    const { asFragment } = customRender(
      <UITools.PopoverCompound.PopoverRoot selected={false} setSelected={noop}>
        <PopoverInsets
          icon="PaddingB"
          insets={['b', 'l', 'r', 't']}
          insetsName="padding"
          isMixed
          translationNameSpace={paddingTranslationNameSpace}
          value="1"
        />
      </UITools.PopoverCompound.PopoverRoot>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
