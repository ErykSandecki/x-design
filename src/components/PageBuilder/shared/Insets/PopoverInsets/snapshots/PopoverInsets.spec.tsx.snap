// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopoverGap snapshots should render PopoverInsets 1`] = `
<DocumentFragment>
  <div
    class="css-1wfopar PopoverItem PopoverItem--light"
    data-test-box=""
    data-test-popover-item="0"
  >
    <div
      class="css-1jnxvbp PopoverItem__checkIcon PopoverItem__checkIcon--light PopoverItem__checkIcon--selected PopoverItem__checkIcon--selected--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
      data-test-icon="check"
      height="12"
      name="Check"
      width="12"
    />
    <div
      class="css-1jnxvbp PopoverItem__icon PopoverItem__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
      data-test-icon="padding-b"
      height="12"
      name="PaddingB"
      width="12"
    />
    <small
      class="css-1jnxvbp PopoverItem__text PopoverItem__text--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
      data-test-text=""
      style="text-align: inherit;"
    >
      Fixed padding (1)
    </small>
  </div>
</DocumentFragment>
`;

exports[`PopoverGap snapshots should render with mixed value 1`] = `
<DocumentFragment>
  <div
    class="css-1wfopar PopoverItem PopoverItem--light"
    data-test-box=""
    data-test-popover-item="0"
  >
    <div
      class="css-1jnxvbp PopoverItem__checkIcon PopoverItem__checkIcon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
      data-test-icon="check"
      height="12"
      name="Check"
      width="12"
    />
    <div
      class="css-1jnxvbp PopoverItem__icon PopoverItem__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
      data-test-icon="padding-b"
      height="12"
      name="PaddingB"
      width="12"
    />
    <small
      class="css-1jnxvbp PopoverItem__text PopoverItem__text--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
      data-test-text=""
      style="text-align: inherit;"
    >
      Fixed padding (1)
    </small>
  </div>
</DocumentFragment>
`;
